### PR TITLE
copy worker-dom files from the new amp-production directory when it exists

### PR DESCRIPTION
**Pull request by erwinmombay**
- remove the copying of "mjs" files into "max.js" files (turning modules into nomodules is a big no no)
- copy the "js" files to "mjs" files to fix the missing "mjs" files for when the module version of amp is loaded (this is OK for now as a temp fix)
- if /dist/amp-production exists in worker-dom then the new directory structure implemented has https://github.com/ampproject/worker-dom/pull/929 is merged and we should copy from the new directories
- make sure to copy source map files as well
- upgrade worker-dom version
- fix import paths to main.mjs

### CODE REVIEW

File: build-system/tasks/extension-helpers.js
Comment (samouri) [inline build-system/tasks/extension-helpers.js:578]: Optional: Do you think it could be simpler with a map instead? e.g.
```js
const filenameMap = {
  'amp-production/worker.js': `amp-script-worker-${version}.js`,
  'amp-production/worker.mjs': `amp-script-worker-${version}.mjs`,
  'amp-production/worker.nodom.js': `amp-script-worker-nodom-${version}.js`,
  'amp-production/worker.nodom.mjs': `amp-script-worker-nodom-${version}.mjs`,

  'amp-debug/worker.js': `amp-script-worker-${version}.max.js`,
  'amp-debug/worker.mjs': `amp-script-worker-${version}.max.mjs`,
  'amp-debug/worker.nodom.js': `amp-script-worker-nodom-${version}.max.js`,
  'amp-debug/worker.nodom.mjs': `amp-script-worker-nodom-${version}.max.mjs`,
}
```

**Review by samouri — APPROVED** [review]
LGTM. If we can get the worker-dom release done today, then I think I'd vote to hold off merging this PR until then.

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:613]: This code is pretty substantial in size and complexity and should probably be moved into a separate function with its own JSDoc.

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [reply build-system/tasks/extension-helpers.js:578]: Agree, this would make the code more maintainable.

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:570]: Nit: Use the established logging standard in our compilation code. For example:

https://github.com/ampproject/amphtml/blob/b5dea36e0b8bd012585d50839766a084f99a3685/build-system/tasks/extension-helpers.js#L247-L250

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:22]: Another `path` clash rears its head ☹️ 

For consistency with other parts of the codebase, we call this `pathModule` instead of `pathmod`? A future cleanup is forthcoming.

File: build-system/tasks/extension-helpers.js
Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js:578]: i actually made this a map originally, if that is preferred i will do that. i simply switched to this since everything was copyable

Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js]: done.

File: build-system/tasks/extension-helpers.js
Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js]: done.

File: build-system/tasks/extension-helpers.js
Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js]: removed since we enumerate the paths now

**Comment by erwinmombay** [comment]
@rsimha @samouri PTAL. thank you

File: build-system/tasks/extension-helpers.js
Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js]: done.

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:647]: Since `dir` and `targetDir` are common across all entries, how about removing them from the map and changing this line as follows?

```js
await fs.copy(`${dir}/${src}`), `${targetDir}/${dest}`);
```

**Review by rsimha — APPROVED** [review]

File: build-system/tasks/extension-helpers.js
Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js]: done.

**Comment by erwinmombay** [comment]
@samouri ill hold off on merging this until you say its good

**Comment by samouri** [comment]
> @samouri ill hold off on merging this until you say its good

@erwinmombay: should be able to bump versions to 27.1 now and delete one of the code branches.

**Comment by rsimha** [comment]
> @erwinmombay: should be able to bump to [version to 27.1](https://github.com/ampproject/worker-dom/releases/tag/v0.27.1) now and delete one of the code branches.

You can merge https://github.com/ampproject/amphtml/pull/30096 to upgrade `worker-dom`

**Comment by erwinmombay** [comment]
@rsimha @samouri i went ahead and cherry picked the changes from https://github.com/ampproject/amphtml/pull/30096 and fixed the import paths to main.mjs. PTAL. thanks

**Review by samouri — APPROVED** [review]

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:575]: Sorry about making you change this again, but I pointed you to the wrong logging example earlier.

image

Could we change this as follows so the step can be timed?

At the start of this function:
```js
  const startTime = Date.now(); 
```

At the end of this function:
```js
    endBuildStep(
      '@ampproject/worker-dom resources',
      'worker.js, worker.nodom.js',
      startTime
    );
```

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:568]: Nit: Add JSDoc for this function.

**Review by rsimha — APPROVED** [review]
Glad the version upgrade worked! LGTM after a couple cosmetic changes requested below.

**Review by kristoferbaxter — APPROVED** [review]
LGTM with the small requests @rsimha asked for.

File: build-system/tasks/extension-helpers.js
Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js:575]: its np. ill make the change. appreciate the thoroughness 👍

File: build-system/tasks/extension-helpers.js
Comment (erwinmombay) [reply build-system/tasks/extension-helpers.js:574]: done.

File: build-system/tasks/extension-helpers.js
Comment (rsimha) [inline build-system/tasks/extension-helpers.js:581]: This can now be deleted.

Comment (rsimha) [reply build-system/tasks/extension-helpers.js:581]: @erwinmombay Can you delete the duplicate logging in a follow up?

Comment (rsimha) [reply build-system/tasks/extension-helpers.js:581]: Sent out #30373 for this.

**Pull request merged**
